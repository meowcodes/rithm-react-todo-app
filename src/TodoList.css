.TodoList-form {
    width: 100vw;
}
.TodoList-list {
    display: inline-block;
    width: 60vw;
}
.TodoList-edit {
    display: inline-block;
    width: 40vw;
}