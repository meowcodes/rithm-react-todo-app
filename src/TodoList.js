import React, { Component } from 'react';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';
import uuid from 'uuid/v4';


class TodoList extends Component {
    // todoArr
    constructor(props) {
        super(props);
        this.state = {
            todoArr: []
        }

        this.createTodo = this.createTodo.bind(this)
        this.deleteTodo = this.deleteTodo.bind(this)
    }

    // create Todo
    createTodo(todo){
        let newTodo = {...todo, id: uuid()}
        this.setState( state => ({
            todoArr: [...state.todoArr, newTodo]
        }));
    }

    // delete Todo
    deleteTodo(id){
        this.setState(state => ({
            todoArr: state.todoArr.filter(todo => todo.id !== id)
        }));
    }

    render() {
        return (
            <div>
                <NewTodoForm notifySubmit={ this.createTodo }/>
                <ul>
                    {this.state.todoArr.map( todo => (
                        <Todo 
                            text={ todo.text }
                            key={ todo.id }
                            notifyDelete={ () => this.deleteTodo(todo.id) }
                        />
                    ))}
                </ul>
            </div>
        )
    }
}

export default TodoList;