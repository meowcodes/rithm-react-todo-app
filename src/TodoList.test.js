import React from "react";
import { mount } from "enzyme";
import TodoList from "./TodoList";
import toJson from "enzyme-to-json";

let wrapper;

beforeEach(function () {
    wrapper = mount(<TodoList />);
})

it("renders without crashing", function () {
    mount(<TodoList />);
});

it("matches snapshot", function () {
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});

it("adds todo items", function() {
    wrapper.setState({
        todoArr: [{
            text: "need to test",
            id: 1
        }]
    });

    expect(wrapper.state().todoArr).toEqual(
        [{ text: "need to test", id: 1}]
    );
    
    wrapper.instance().createTodo({
        text: "need to test more"
    });

    expect(wrapper.state().todoArr).toEqual(
        [
            { text: "need to test", id: 1},
            { text: "need to test more", id: expect.any(String)}
        ]
    );
});

it("renders a form and successfully creates a todo", function() {
    const textInput = wrapper.find("#text");
    textInput.instance().value = "need to test";
    textInput.simulate("change");

    const form = wrapper.find("form");
    form.simulate("submit");

    expect(wrapper.state().todoArr.length).toEqual(1);

    expect(
        wrapper.find("p").last().text()
    ).toEqual("need to test")
});

// deletes when delete btn clicked