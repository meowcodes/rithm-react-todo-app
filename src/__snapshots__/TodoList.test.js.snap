// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<TodoList>
  <div>
    <NewTodoForm
      notifySubmit={[Function]}
    >
      <form
        onSubmit={[Function]}
      >
        <label
          htmlFor="text"
        >
          Todo
        </label>
        <input
          id="text"
          name="text"
          onChange={[Function]}
          value=""
        />
        <button>
          Create Todo
        </button>
      </form>
    </NewTodoForm>
    <ul />
  </div>
</TodoList>
`;
